apps/passport-server/src/services/userService.ts

```diff
@@ -322,6 +355,65 @@ export class UserService {
 
     return user;
   }
+
+  /**
+   * Updates the version of the legal terms the user agrees to
+   */
+  public async handleAgreeTerms(
+    serializedPCD: SerializedPCD<SemaphoreSignaturePCD>
+  ): Promise<AgreeTermsResult> {
+    const pcd = await SemaphoreSignaturePCDPackage.deserialize(
+      serializedPCD.pcd
+    );
+    if (await SemaphoreSignaturePCDPackage.verify(pcd)) {
+      const parsedPayload = AgreedTermsSchema.safeParse(
+        JSON.parse(pcd.claim.signedMessage)
+      );
+
+      if (!parsedPayload.success) {
+        return {
+          success: false,
+          error: "Invalid terms specified"
+        };
+      }
+
+      const payload = parsedPayload.data;
+      const user = await fetchUserByCommitment(
+        this.context.dbPool,
+        pcd.claim.identityCommitment
+      );
+      if (!user) {
+        return {
+          success: false,
+          error: "User does not exist"
+        };
+      }
+
+      if (payload.version < UNREDACT_TICKETS_TERMS_VERSION) {
+        await upsertUser(this.context.dbPool, {
+          ...user,
+          terms_agreed: payload.version
+        });
+      } else {
```

Comment:
I can't think of a thing that we want to do if and only if the version being agreed to is exactly this one. What case is this meant to cover?

Reply:
I've changed how this part works:

If the user is agreeing to terms which permit unredaction, and has not previously done so, unredact the tickets. Otherwise, just record whatever they are agreeing to. This could be simpler if we just assume that this never gets called except to agree to the latest version though.